ksql-engine/src/main/java/io/confluent/ksql/services/ServiceContextFactory.java

```diff
@@ -48,19 +48,19 @@ public static ServiceContext create(
       final KafkaClientSupplier kafkaClientSupplier,
       final Supplier<SchemaRegistryClient> srClientFactory,
       final ConnectClient connectClient,
-      final SimpleKsqlClient ksqlClient
+      final Supplier<SimpleKsqlClient> ksqlClientSupplier
   ) {
-    final Admin adminClient = kafkaClientSupplier.getAdmin(
-        ksqlConfig.getKsqlAdminClientConfigProps()
-    );
+
+    ClientContextFactory clientsContext = new ClientContextFactory(
```

Comment:
So the decision here is to still create the ClientContextFactory every request and hope its cheap? 

I am fine with this for now. But we will have revisit all this in the context of caching in general for RBAC..

Reply:
This should be cheap as it is just keeping the suppliers. No object is being created, until actually needed. And because of memoization, it is created only once per request. For Pull queries, none should be created.